### ISSUE TITLE
Add init() to the application context interface

### ISSUE THREAD

**Issue by jurajsipos** [nestjs/nest#1592]
Hi guys,

for purpose of unit testing & integration testing, is there any way of having lifecycle events executed with `TestingModule` without calling `createApplication`? It would be great if you wouldn't have to start express server for each unit test and would have `Init` and `Destroy` lifecycle events properly executed after compilation of a testing module, or after calling some kind of `init()`.

<pre><code>
[ ] Regression 
[ ] Bug report
[x ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior
Lifecycle events not being executed within `TestModule` without creating express application.

## Expected behavior
Lifecycle events get executed within `TestModule` after compilation or after calling some kind of `init()` or `createApplicationContext()`.

## Minimal reproduction of the problem with instructions

```
  beforeEach(async () => {
    const module = await Test.createTestingModule({
        imports: [CommonModule],
        providers: [CatsService],
      }).compile();

    catsService = module.get<CatsService>(CatsService);
  });
```
In this example, if `CommonModule` has lots of services which need proper initialisation and cleanup in `OnModuleInit` & `OnModuleDestroy` handlers, we are not able to `init()` the module without calling `createApplication` which will start express server.
## Environment

<pre><code>
Nest version: 5.7.1

 
For Tooling issues:
- Node version: 10.14.0
- Platform:  Mac

Others:

</code></pre>

Many thanks

**Comment by kamilmysliwiec** [comment]
`close()` should be available. We need to add `init()` though

**Comment by kamilmysliwiec** [comment]
Published as 6.3.0
